hydrolib/core/io/rr/network/parser.py

```diff
@@ -0,0 +1,74 @@
+from pathlib import Path
+from warnings import warn
+
+from hydrolib.core.utils import get_list_index_safely
+
+
+class NodeFileParser:
+    """A parser for topology node files."""
+
+    @staticmethod
+    def parse(path: Path):
+        """Parses a topology node file to a dictionary.
+
+        Args:
+            path (Path): Path to the topology node file.
+        """
+
+        if not path.is_file():
+            warn(f"File: `{path}` not found, skipped parsing.")
+            return {}
+
+        with open(path) as file:
+            parts = file.read().split()
+
+        nodes = []
+
+        key_start = "NODE"
+        key_end = "node"
+
+        length_parts = len(parts)
+
+        index = 0
+        while index < length_parts:
```

Comment:
Is it not safer for (large) files to replace the above ```file.read()``` and this while loop here by:
```
file = open(path)
# ...
for line in file:
    # ...
```

Reply:
line-by-line reading is sufficient, because RR requires each node record to be on a single "NODE...node" line.

Reply:
Ok, good to know, I assumed that the node and NODE were leading in terms of when a node starts and ends. Didn't think it was invalid to have a multiple line node.
Will change it!